Tasks/TwineAuthenticateV0/TwineAuthenticateMain.ts

```diff
@@ -0,0 +1,120 @@
+import * as fs from "fs";
+import * as ini from "ini";
+import * as os from "os";
+import * as path from "path";
+import * as util from "util";
+import * as telemetry from "utility-common/telemetry";
+import * as tl from "vsts-task-lib";
+import * as auth from "./Authentication";
+import * as utils from "./Utilities";
+
+ // tslint:disable-next-line:max-classes-per-file
+export class Repository
+{
+    public repository: string;
+    public username: string;
+    public password: string;
+
+    constructor(repository: string, username: string, password: string)
+     {
+        this.repository = repository;
+        this.username = username;
+        this.password = password;
+    }
+}
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, "task.json"));
+    try {
+        // Local feeds
+        const internalFeeds = await auth.getInternalAuthInfoArray("feedList");
+        // external service endpoints
+        const externalEndpoints = await auth.getExternalAuthInfoArray("externalSources");
+        // combination of both internal and external
+        const newEndpointsToAdd = internalFeeds.concat(externalEndpoints);
+
+        let pypircPath = utils.getPypircPath();
+
+        // Case when there are multiple twine auth tasks in a build
+        if (tl.exist(pypircPath)) {
+            // merge two tasks
+            tl.debug(tl.loc("Info_StartParsingExistingPypircFile", pypircPath));
+            let fileContent = ini.parse(fs.readFileSync(pypircPath, "utf-8"));
+
+            // Adding new endpoints to already existing .pypirc file.
+            for (const entry of newEndpointsToAdd){
+                if (entry.packageSource.feedName in fileContent){
+                    // Don't fail if new feed is from same collection
+                    if (entry.packageSource.isInternalSource){
+                        tl.warning(tl.loc("Warning_DuplicateEntryForInternalFeed",
+                        entry.packageSource.feedName));
+                        continue;
+                    }
+                    else{
+                        // Hard fail if there is a name collision from service endpoint
+                        throw new Error(tl.loc("Error_DuplicateEntryForExternalFeed",
+                        entry.packageSource.feedName));
+                    }
```

Comment:
Discussed offline: hard fail on both cases for now; otherwise, the order in which the entries are added (external first vs external second) would change what happens here